src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -574,8 +577,28 @@ private Task WriteHexInt32Async(int value)
             }
         }
 
-        private HttpContentWriteStream CreateRequestContentStream(HttpRequestMessage request)
+        public Task<HttpResponseMessage> SendWithNtProxyAuthAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (_pool.UsingProxy && _pool.ProxyCredentials != null)
+            {
+                return AuthenticationHelper.SendWithNtProxyAuthAsync(request, _pool.ProxyUri, _pool.ProxyCredentials, this, cancellationToken);
+            }
+
+            return SendAsyncCore(request, cancellationToken);
+        }
+
+        public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, bool doRequestAuth, CancellationToken cancellationToken)
```

Comment:
There are now a bunch of methods that begin with SendAsync spread around, and I'm having a little trouble keeping straight which does what.  Could you add a comment somewhere explaining the flow, or individual comments explaining in what circumstance each of these is called?

Reply:
Sure